apps/web/src/app/vault/guards/setup-extension-redirect.guard.ts

```diff
@@ -0,0 +1,106 @@
+import { inject } from "@angular/core";
+import { CanActivateFn, Router } from "@angular/router";
+import { Observable, firstValueFrom, map } from "rxjs";
+
+import { VaultProfileService } from "@bitwarden/angular/vault/services/vault-profile.service";
+import { Account, AccountService } from "@bitwarden/common/auth/abstractions/account.service";
+import { FeatureFlag } from "@bitwarden/common/enums/feature-flag.enum";
+import { ConfigService } from "@bitwarden/common/platform/abstractions/config/config.service";
+import { Utils } from "@bitwarden/common/platform/misc/utils";
+import {
+  SETUP_EXTENSION_DISMISSED_DISK,
+  StateProvider,
+  UserKeyDefinition,
+} from "@bitwarden/common/platform/state";
+
+import { WebBrowserInteractionService } from "../services/web-browser-interaction.service";
+
+export const SETUP_EXTENSION_DISMISSED = new UserKeyDefinition<boolean>(
+  SETUP_EXTENSION_DISMISSED_DISK,
+  "setupExtensionDismissed",
+  {
+    deserializer: (dismissed) => dismissed,
+    clearOn: [],
+  },
+);
+
+export const setupExtensionRedirectGuard: CanActivateFn = async () => {
+  const router = inject(Router);
+  const configService = inject(ConfigService);
+  const accountService = inject(AccountService);
+  const vaultProfileService = inject(VaultProfileService);
+  const stateProvider = inject(StateProvider);
+  const webBrowserInteractionService = inject(WebBrowserInteractionService);
+
+  const isMobile = Utils.isMobileBrowser;
+
+  const endUserFeatureEnabled = await configService.getFeatureFlag(
+    FeatureFlag.PM19315EndUserActivationMvp,
+  );
+
+  // The extension page isn't applicable for mobile users, do not redirect them.
+  // Include before any other checks to avoid unnecessary processing.
+  if (!endUserFeatureEnabled || isMobile) {
+    return true;
+  }
+
+  const currentAcct$: Observable<Account | null> = accountService.activeAccount$;
+  const currentAcct = await firstValueFrom(currentAcct$);
```

Comment:
⛏️ (non-blocking): We don't need a separate `currentAcct$` variable. 

```suggestion
  const currentAcct = await firstValueFrom(accountService.activeAccount$);
```

Reply:
[2101a86](https://github.com/bitwarden/clients/pull/15641/commits/2101a8691b53c6d7bb4394cbaab145a167ad0697)